### PR TITLE
Fix INITSSLOT refs

### PR DESCRIPTION
**Pull request by roman-khimov**
### Problem

#2501

### Solution

Aquí.

### CODE REVIEW

File: pkg/vm/slot.go [hunk 1]
```diff
@@ -23,13 +23,8 @@ func (s slot) Set(i int, item stackitem.Item, refs *refCounter) {
 	if s[i] == item {
 		return
 	}
-	old := s[i]
+	refs.Remove(s[i])
 	s[i] = item
-	if old != nil {
-		refs.Remove(old)
-	} else {
-		*refs-- // Not really existing, but counted Null element.
```
Comment (AnnaShaleva) [inline pkg/vm/slot.go:31]: But how about this Null element?

Comment (roman-khimov) [reply pkg/vm/slot.go:31]: `refs.Remove(nil)` which works just fine, decrementing the counter.

Comment (AnnaShaleva) [reply pkg/vm/slot.go:31]: Are you sure that `refs.Remove(nil)` decrements the counter?

**Review by fyrchik — APPROVED** [review]
Me parece bien!

File: pkg/vm/slot.go (hunk 1 above)
Comment (roman-khimov) [reply pkg/vm/slot.go:31]: Yeah, why not?

Comment (AnnaShaleva) [reply pkg/vm/slot.go:31]: Ugh, I’ve misread the code, everything is OK here, but tests need to be fixed.

Comment (roman-khimov) [reply pkg/vm/slot.go:31]: Fixed.

**Review by AnnaShaleva — APPROVED** [review]

**Pull request merged**